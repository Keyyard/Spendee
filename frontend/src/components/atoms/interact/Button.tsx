import React from 'react';
import { TouchableOpacity, Text, TouchableOpacityProps } from 'react-native';

interface ButtonProps extends TouchableOpacityProps {
  title: string;
  variant?: 'primary' | 'danger' | 'success';
  className?: string;
}

const baseStyles = 'p-4 rounded-lg';
const variants = {
  primary: 'bg-blue-600',
  danger: 'bg-red-600',
  success: 'bg-green-500',
};

export const Button: React.FC<ButtonProps> = ({ title, variant = 'primary', className = '', ...props }) => (
  <TouchableOpacity {...props} className={`${baseStyles} ${variants[variant]} ${className}`}>
    <Text className="text-white text-center font-bold">{title}</Text>
  </TouchableOpacity>
);

export default Button;
