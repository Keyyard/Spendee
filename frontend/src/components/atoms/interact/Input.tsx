import React from 'react';
import { TextInput, TextInputProps } from 'react-native';

interface InputProps extends TextInputProps {
  className?: string;
}

const baseStyles = 'border border-gray-300 p-4 rounded-lg';

export const Input: React.FC<InputProps> = ({ className = '', ...props }) => (
  <TextInput {...props} className={`${baseStyles} ${className}`} />
);

export default Input;
