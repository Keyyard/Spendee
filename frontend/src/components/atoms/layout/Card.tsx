import React from 'react';
import { View, ViewProps } from 'react-native';

interface CardProps extends ViewProps {
  className?: string;
  children: React.ReactNode;
}

export const Card: React.FC<CardProps> = ({ className = '', children, ...props }) => (
  <View {...props} className={`bg-white rounded-2xl p-5 shadow-sm ${className}`}>{children}</View>
);

export default Card;
