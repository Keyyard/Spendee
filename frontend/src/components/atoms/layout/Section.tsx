import React from 'react';
import { View, ViewProps } from 'react-native';

interface SectionProps extends ViewProps {
  className?: string;
  children: React.ReactNode;
}

export const Section: React.FC<SectionProps> = ({ className = '', children, ...props }) => (
  <View {...props} className={`mb-4 ${className}`}>{children}</View>
);

export default Section;
