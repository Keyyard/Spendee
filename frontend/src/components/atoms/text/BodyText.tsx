import React from 'react';
import { Text, TextProps } from 'react-native';

interface BodyTextProps extends TextProps {
  className?: string;
  children: React.ReactNode;
}

export const BodyText: React.FC<BodyTextProps> = ({ className = '', children, ...props }) => (
  <Text {...props} className={`text-base text-gray-700 ${className}`}>{children}</Text>
);

export default BodyText;
