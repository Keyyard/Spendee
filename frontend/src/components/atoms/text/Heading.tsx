import React from 'react';
import { Text, TextProps } from 'react-native';

interface HeadingProps extends TextProps {
  level?: 1 | 2 | 3 | 4 | 5 | 6;
  className?: string;
  children: React.ReactNode;
}

const sizes = {
  1: 'text-3xl font-bold',
  2: 'text-2xl font-bold',
  3: 'text-xl font-semibold',
  4: 'text-lg font-semibold',
  5: 'text-base font-semibold',
  6: 'text-sm font-semibold',
};

export const Heading: React.FC<HeadingProps> = ({ level = 2, className = '', children, ...props }) => (
  <Text {...props} className={`${sizes[level]} ${className}`}>{children}</Text>
);

export default Heading;
