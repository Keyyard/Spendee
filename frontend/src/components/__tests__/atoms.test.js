import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../atoms/interact/Button';
import { Input } from '../atoms/interact/Input';
import { Card } from '../atoms/layout/Card';
import { Section } from '../atoms/layout/Section';
import { Heading } from '../atoms/text/Heading';
import { BodyText } from '../atoms/text/BodyText';


describe('Atoms Components (renderer)', () => {
  it('renders Button', () => {
    const tree = renderer.create(<Button title="Test Button" onPress={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Input', () => {
    const tree = renderer.create(<Input placeholder="Type here" value="" onChangeText={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Card with children', () => {
    const tree = renderer.create(
      <Card><BodyText>Card Content</BodyText></Card>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Section with children', () => {
    const tree = renderer.create(
      <Section><BodyText>Section Content</BodyText></Section>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Heading at different levels', () => {
    const tree = renderer.create(
      <>
        <Heading level={1}>Heading 1</Heading>
        <Heading level={2}>Heading 2</Heading>
      </>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders BodyText', () => {
    const tree = renderer.create(<BodyText>Body Content</BodyText>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
