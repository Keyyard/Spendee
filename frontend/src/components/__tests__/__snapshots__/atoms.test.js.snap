// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms Components (renderer) renders BodyText 1`] = `
<Text
  className="text-base text-gray-700 "
>
  Body Content
</Text>
`;

exports[`Atoms Components (renderer) renders Button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    className="text-white text-center font-bold"
  >
    Test Button
  </Text>
</View>
`;

exports[`Atoms Components (renderer) renders Card with children 1`] = `
<View
  className="bg-white rounded-2xl p-5 shadow-sm "
>
  <Text
    className="text-base text-gray-700 "
  >
    Card Content
  </Text>
</View>
`;

exports[`Atoms Components (renderer) renders Heading at different levels 1`] = `
[
  <Text
    className="text-3xl font-bold "
  >
    Heading 1
  </Text>,
  <Text
    className="text-2xl font-bold "
  >
    Heading 2
  </Text>,
]
`;

exports[`Atoms Components (renderer) renders Input 1`] = `
<TextInput
  className="border border-gray-300 p-4 rounded-lg "
  onChangeText={[Function]}
  placeholder="Type here"
  value=""
/>
`;

exports[`Atoms Components (renderer) renders Section with children 1`] = `
<View
  className="mb-4 "
>
  <Text
    className="text-base text-gray-700 "
  >
    Section Content
  </Text>
</View>
`;
